demos/shadow/shadow_demo_main/shadow_demo_main.c

```diff
@@ -148,6 +148,30 @@
  */
 #define SHADOW_REPORTED_JSON_LENGTH    ( sizeof( SHADOW_REPORTED_JSON ) - 3 )
 
+/**
+ * @brief The maximum number of times to run the loop in this demo.
+ */
+#ifndef SHADOW_MAX_DEMO_COUNT
+    #define SHADOW_MAX_DEMO_COUNT    ( 3 )
+#endif
+
+/**
+ * @brief Time in seconds to wait between retries of the demo loop if
+ * demo loop fails.
+ */
+#define DELAY_BETWEEN_DEMO_ITERATIONS_S                 ( 5 )
+
+/**
+ * @brief JSON key for response code that indicates the type of error in
+ * the error document received on topic `delete/rejected`.
```

Comment:
Based on the API doc for [Device Shadow](https://docs.aws.amazon.com/iot/latest/developerguide/device-shadow-mqtt.html#delete-rejected-pub-sub-topic)
```suggestion
 * the error document received on topic `/delete/rejected`.
```